Terminal.Gui/Views/Scroll.cs

```diff
@@ -0,0 +1,387 @@
+﻿namespace Terminal.Gui;
+
+/// <summary>
+///     Represents the "inside part" of a scroll bar, minus the arrows.
+/// </summary>
+public class Scroll : View
+{
+    /// <inheritdoc/>
+    public Scroll ()
+    {
+        WantContinuousButtonPressed = true;
+        CanFocus = false;
+        Orientation = Orientation.Vertical;
+        Width = Dim.Auto (DimAutoStyle.Content, 1);
+        Height = Dim.Auto (DimAutoStyle.Content, 1);
+
+        _slider = new ()
+        {
+            Id = "slider",
+            Width = Dim.Auto (DimAutoStyle.Content),
+            Height = Dim.Auto (DimAutoStyle.Content)
+        };
+        Add (_slider);
+
+        Added += Scroll_Added;
+        Removed += Scroll_Removed;
+        Initialized += Scroll_Initialized;
+        DrawContent += Scroll_DrawContent;
+        MouseEvent += Scroll_MouseEvent;
+        _slider.DrawContent += Scroll_DrawContent;
+        _slider.MouseEvent += Slider_MouseEvent;
+    }
+
+    private readonly View _slider;
+
+    private int _lastLocation = -1;
+
+    private bool _wasSliderMouse;
```

Comment:
> I don't fully understand what this is used for and why it and `Parent_LayoutComplete` are needed.
> 

The `_wasSliderMouse` field is necessary because I need to distinguish whether the `Position` was changed due to iteration outside the slider or not. When the `Position` changes, the `AdjustSlider` method is called and may, due to rounding in the calculation, undo the slider's new location. That's why I need the `_wasSliderMouse` field to prevent the `AdjustSlider` method from being invalidated.

> This feels like you are working around some deficiency in the library we should fix at a deeper level because having a subview that can be moved via the mouse is a pretty common thing.
> 
> I don't have any suggestions on how to fix, but it makes me curious.

As for `Parent_LayoutComplete` it may not be necessary and I used it more because of some resizing and ensuring the recalculation of new values. I can try removing it and check if the `Scroll` automatically adjusts to the new values, based on the new superview dimension.